src/util/JSONClientSingleton.js

```diff
@@ -17,12 +17,32 @@ class JSONClientSingleton {
     }
     this.throttleTimeout_ = null;
     this.debounceErrors_ = false;
+    this.isConnected = false;
+    this.eventHandler = function(type, opt_data) {};
     singletonInstance = this;
     return singletonInstance;
   }
+  
+  setEventHandler(eventHandler) {
+    this.eventHandler = eventHandler;
+  }
+
+  setServerUrl(url, login, password) {
```

Comment:
Please use opt_login, opt_password (a convention to indicate that they are optional).